### PR TITLE
Fix pthread

### PR DESCRIPTION
**Pull request by jpakkane**
Same as #4286 but with a fix to the macOS test failure.

### CODE REVIEW

File: run_unittests.py [hunk 1]
```diff
@@ -3270,13 +3270,23 @@ def test_pkgconfig_gen_deps(self):
         self.assertEqual(sorted(out), sorted(['libfoo >= 1.0']))
 
         out = self._run(cmd + ['--cflags-only-other']).strip().split()
+        if is_osx():
+            # macOS does not use -pthread so add it in to make
+            # the test logic a bit easier to implement.
+            out += ['-pthread']
```
Comment (ePirat) [inline run_unittests.py]: I can see this going to create a very confusing output on failures suggesting there is `-pthread` present even though it is not by meson but from the test logic itself.

Comment (nirbheek) [reply run_unittests.py]: I like the approach in https://github.com/mesonbuild/meson/pull/4305 better: `[] if is_osx() else ['-pthread']`.

**Comment by SoapGentoo** [comment]
In general you should adapt the invariant to what you're testing, not the output. Hence why I used the ternary clause for that.

File: run_unittests.py (hunk 1 above)
Comment (dcbaker) [reply run_unittests.py]: I think this needs to be `if_osx() or if_haiku()`, right?

**Comment by jpakkane** [comment]
I moved all the ugliness to its own method so it is isolated and in one place.

File: run_unittests.py [hunk 2]
```diff
@@ -3290,6 +3290,11 @@ def test_pkgconfig_gen_deps(self):
         out = self._run(cmd + ['--print-requires-private']).strip().split('\n')
         self.assertEqual(sorted(out), sorted(['libexposed', 'libfoo >= 1.0', 'libhello']))
 
+    def check_pkg_flags_are_same(self, output, expected):
+        if is_osx():
```
Comment (dcbaker) [inline run_unittests.py]: I just tested, haiku fails this test as well, can we add `is_haiku()` here? Then this version looks good, I'll close mine.

**Pull request merged**